### PR TITLE
Odometry - estimate mu

### PR DESCRIPTION
**Pull request by james-ward**

### CODE REVIEW

File: swervedrive/icr/kinematicmodel.py [hunk 1]
```diff
@@ -132,3 +129,57 @@ def reconfigure_wheels(self, beta_d: np.ndarray, beta_e: np.ndarray):
...
+        xi_dot = mu_e * np.array([[lmda_e[1]], [-lmda_e[0]], [lmda_e[2]]])  # Eq (2)
+        theta = self.xi[0, 2]
+        m3 = np.array(
+            [
+                [np.cos(theta), -np.sin(theta), 0],
+                [np.sin(theta), np.cos(theta), 0],
+                [0, 0, 1],
+            ]
+        )
+        np.matmul(m3, xi_dot)
```
Comment (auscompgeek) [inline swervedrive/icr/kinematicmodel.py]: Hm, this seems to be unused.

Comment (james-ward) [reply swervedrive/icr/kinematicmodel.py]: Good catch.
Fixed in latest commit.

File: swervedrive/icr/kinematicmodel.py [hunk 2]
```diff
@@ -132,3 +129,56 @@ def reconfigure_wheels(self, beta_d: np.ndarray, beta_e: np.ndarray):
...
+        Find the rotational position of the robot about the ICR.
+        :param phi_dot: array of angular velocities of the wheels.
+        :param lmda_e: the estimate of the ICR in h-space.
+        :returns: the estimate of mu (float).
+        """
+        # this requires solving equation (22) from the control paper, i think
+        # we may need to look into whether this is valid for a system with no
+        # wheel coupling
+        lmda_e = np.reshape(lmda_e, (len(lmda_e), 1))
+        phi_dot = np.reshape(phi_dot, (len(phi_dot), 1))
```
Comment (ArthurAllshire) [inline swervedrive/icr/kinematicmodel.py]: This is not a very safe way of reshaping. Can lead to issues like:
```
(ins)>>> import numpy as np
(ins)>>> a = np.array([[3, 4, 5]])
(ins)>>> a
array([[3, 4, 5]])
(ins)>>> len(a)
1
```
We should be using (-1, 1) instead.

**Review by ArthurAllshire — DISMISSED** [review]
I will change the array reshaping thing and push onto this PR.

**Review by ArthurAllshire — APPROVED** [review]

**Pull request merged**